wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/ConversationMessageActionController.swift

```diff
@@ -88,38 +100,50 @@ final class ConversationMessageActionController {
     func canPerformAction(action: MessageAction) -> Bool {
         switch action {
         case .copy:
-            message.canBeCopied
+            return message.canBeCopied
         case .digitallySign:
-            message.canBeDigitallySigned
+            return message.canBeDigitallySigned
         case .reply:
-            message.canBeQuoted
+            return message.canBeQuoted
         case .openDetails:
-            message.areMessageDetailsAvailable
+            return message.areMessageDetailsAvailable
         case .edit:
-            message.canBeEdited
+            return message.canBeEdited
         case .delete:
-            message.canBeDeleted
+            return message.canBeDeleted
         case .save:
-            message.canBeSaved
+            return message.canBeSaved
         case .cancel:
-            message.canCancelDownload
+            return message.canCancelDownload
         case .download:
-            message.canBeDownloaded
+            return message.canBeDownloaded
         case .resend:
-            message.canBeResent
+            return message.canBeResent
         case .showInConversation:
-            context == .collection
+            return context == .collection
         case .sketchDraw,
              .sketchEmoji:
-            message.isImage
+            return message.isImage
         case .react:
-            message.canAddReaction
+            return message.canAddReaction
         case .visitLink:
-            message.canVisitLink
+            return message.canVisitLink
+        case .collapse:
+            guard let selfUserId,
+                  let isCollapsed,
+                  PrivateUserDefaults<CollapseKey>(userID: selfUserId).bool(forKey: .collapseOwnMessages),
+                  !isCollapsed else {
+                return false
+            }
+
+            let isOfSupportedMessageTypeToCollapse = message.isFile || message.isAudio || message.isVideo || message
```

Comment:
nitpic: name variable
```suggestion
            let messageSupportsCollapsing = message.isFile || message.isAudio || message.isVideo || message
```

Reply:
fixed